### PR TITLE
[wip] added tests for api

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: powerdns/tests/test_api.py [hunk 1]
```diff
@@ -54,3 +54,76 @@ def test_record_filters_by_ip_and_type(self):
...
+        domain = DomainFactory(name='example.com', owner=self.regular_user2)
+        data = {
+            'type': 'cname'.upper(),
+            'domain': '/api/domains/' + str(domain.id) + '/',
+            'name': 'example.com',
+            'content': '192.168.0.1',
+        }
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        data = json.loads(response.content.decode('utf8'))
```
Comment (szok) [inline powerdns/tests/test_api.py]: response.data :)

File: powerdns/tests/test_api.py [hunk 2]
```diff
@@ -54,3 +54,76 @@ def test_record_filters_by_ip_and_type(self):
...
+            'type': 'cname'.upper(),
+            'domain': '/api/domains/' + str(domain.id) + '/',
+            'name': 'example.com',
+            'content': '192.168.0.1',
+        }
+        response = self.client.post(url, data, format='json')
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+    def test_regular_user_owns_domain(self):
+        self.client = APIClient()
```
Comment (szok) [inline powerdns/tests/test_api.py]: move to setUp method :)

File: powerdns/tests/test_api.py [hunk 3]
```diff
@@ -54,3 +54,76 @@ def test_record_filters_by_ip_and_type(self):
...
+from rest_framework import status
+from rest_framework.test import APITestCase
+from powerdns.models.powerdns import Record
+from powerdns.tests.utils import DomainFactory, user_client
+from powerdns.models.requests import RecordRequest
+import json
+class TestAddingRecords(BaseApiTestCase):
+    def setUp(self):
+        super().setUp()
+        self.regular_user1 = get_user_model().objects.create_superuser(
```
Comment (szok) [inline powerdns/tests/test_api.py]: create_user for regular_user

File: powerdns/views.py [hunk 4]
```diff
@@ -65,13 +68,70 @@ class DomainViewSet(OwnerViewSet):
...
+        serializer.is_valid(raise_exception=True)
+        self.perform_create(serializer)
+
+        from rest_framework import status
+        from rest_framework.response import Response
+        from django.conf import settings
+        auto_accept = (
+            self.request.user.is_superuser or
+            self.request.user == serializer.instance.domain.owner or
+            serializer.instance.domain.name in settings.FORCE_AUTO_ACCEPT_DOMAINS
```
Comment (szok) [inline powerdns/views.py:116]: maybe add .lower() to str

File: powerdns/tests/test_api.py [hunk 5]
```diff
@@ -80,3 +80,89 @@ def test_no_access_to_not_owned_domain(self):
...
+from powerdns.models.powerdns import Record
+from powerdns.tests.utils import DomainFactory, user_client
+from powerdns.models.requests import RecordRequest
+from django.core.urlresolvers import resolve
+from urllib import parse
+
+
+class TestAddingRecords(BaseApiTestCase):
+    def setUp(self):
+        self.client = APIClient()
```
Comment (szok) [inline powerdns/tests/test_api.py]: It is not necessary, self.client is initialize in BaseApiTestCase

File: powerdns/views.py (hunk 4 above)
Comment (ghost) [inline powerdns/views.py:116]: added force lower on save()

**Comment by mkurek** [comment]
is it related to #79 somehow?

**Comment by ghost** [comment]
yes

**Pull request closed without merging**